workers/workers.go

```diff
@@ -0,0 +1,199 @@
+package workers
+
+import (
+	"time"
+
+	"github.com/serverless/event-gateway-connector/connection"
+	"github.com/serverless/event-gateway-connector/utils"
+	"github.com/serverless/event-gateway-connector/watcher"
+	"go.uber.org/zap"
+)
+
+// worker is the internal representation of the worker process
+type worker struct {
```

Comment:
Overall would be great if we could put public variables/functions/types in the top of the file and private on the bottom. It helps reading the file.